src/main/java/uk/ac/ebi/eva/evaseqcol/service/SeqColLevelOneService.java

```diff
@@ -208,4 +215,28 @@ public List<SeqColExtendedDataEntity<List<Integer>>> constructIntegerListExtData
         return integerListExtendedDataEntities;
     }
 
+    public List<SeqColMetadataEntity> metadataObjectArrayListToMetadataList(List<Object[]> metadataArray) {
+        List<SeqColMetadataEntity> metadataList = new ArrayList<>();
+        for (Object[] metadataElements : metadataArray) {
+            SeqColMetadataEntity metadataEntity = new SeqColMetadataEntity();
+            metadataEntity.setSourceIdentifier((String) metadataElements[0]);
+            metadataEntity.setSourceUrl((String) metadataElements[1]);
+            metadataEntity.setNamingConvention(SeqColEntity.NamingConvention.valueOf(
+                    (String) metadataElements[2]
+            ));
+            metadataEntity.setCreatedOn((Date) metadataElements[3]);
+            metadataList.add(metadataEntity);
+        }
+        return metadataList;
+    }
+
+    public List<SeqColMetadataEntity> getAllMetadata() {
+        List<Object[]> metadataArrayList = repository.findAllMetadata();
+        return metadataObjectArrayListToMetadataList(metadataArrayList);
```

Comment:
Is it worth using something like [TupleTransformer](https://docs.jboss.org/hibernate/orm/6.0/javadocs/org/hibernate/query/TupleTransformer.html) in the repository to do this? I'm not sure how much difference it makes in practice compared to your conversion method.

Reply:
Makes sense, your implementation is pretty neat so let's keep it like that :+1: